akka-http-caching/src/main/scala/akka/http/caching/LfuCache.scala

```diff
@@ -0,0 +1,91 @@
+package akka.http.caching
+
+import java.util.concurrent.{ CompletableFuture, Executor, TimeUnit }
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.Duration
+import scala.concurrent.Future
+import com.github.benmanes.caffeine.cache.{ AsyncCacheLoader, AsyncLoadingCache, Caffeine }
+import akka.http.caching.LfuCache.{ toJavaMappingFunction, dummyLoader }
+
+import scala.compat.java8.FutureConverters._
+import scala.compat.java8.FunctionConverters._
+
+object LfuCache {
+
+  /**
+   * Creates a new [[akka.http.caching.ExpiringLfuCache]] or
+   * [[akka.http.caching.SimpleLfuCache]] instance depending on whether
+   * a non-zero and finite timeToLive and/or timeToIdle is set or not.
+   */
+  def apply[V](
+    maxCapacity:     Int      = 500,
```

Comment:
Let's move sets of configuration parameters to settings classes for easier (= binary compatible) evolution of the API later on.